### PR TITLE
Move connect_arrays() from nest.cpp to connection_manager

### LINKED ISSUE #1960: connect_arrays should be defined in connection_manager instead of nest.cpp
This is a follow-up to #1942.

The function for `connect_arrays()` is defined in `nest.cpp`. This should be refactored, as `nest.cpp` should only contain small wrapper functions passing work on to the various managers. `connect_arrays` in `nest.cpp` should thus refer to a `connect_arrays` in `connection_manager`.

### PR DESCRIPTION
**Pull request by jasperalbers**
This PR fixes #1960.
The work was done my be and @ackurth.

This PR moves the function `connect_arrays` from `nest.cpp` to `connection_manager.cpp`. What remains in `nest.cpp` is a wrapper that simply calls the function from `connection_manager.cpp`. We were unsure of the exact placement of the new function in `connection_manager.cpp`, here we put it behind all occurrences of `connect()`, in line with the order in `nest.cpp`.

### CODE REVIEW

File: nestkernel/connection_manager.cpp [hunk 1]
```diff
@@ -549,6 +549,163 @@ nest::ConnectionManager::connect( const index snode_id,
...
+          }
+          else
+          {
+            auto dd = static_cast< DoubleDatum* >( ( ( *param_dicts[ tid ] )[ param_pointer_pair.first ] ).datum() );
+            ( *dd ) = *param;
+          }
+        }
+
+        connect(
+          *s, target_node, tid, synapse_model_id, param_dicts[ tid ], delay_buffer, weight_buffer );
```
Comment (stinebuu) [inline nestkernel/connection_manager.cpp:684]: I think this is what is causing the problems on Travis, it should be one line.

**Review by stinebuu — COMMENTED** [review]
Thanks! I just have a tiny comment for making Travis happy. :)

**Comment by jougs** [comment]
>  The work was done my be and @ackurth.

Just for the future: You can add co-authors in the commit message for these kind of collaborative works. That way our release-tools and GitHub will automatically pick up all contributors and everyone gets credited correctly. See [here](https://docs.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors) for some more info about this.

While not yet merged, you can of course also change the commit messages of the commits in this PR by rebasing.

**Comment by jasperalbers** [comment]
> > The work was done my be and @ackurth.
> 
> Just for the future: You can add co-authors in the commit message for these kind of collaborative works. That way our release-tools and GitHub will automatically pick up all contributors and everyone gets credited correctly. See [here](https://docs.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors) for some more info about this.
> 
> While not yet merged, you can of course also change the commit messages of the commits in this PR by rebasing.

Thanks for the reference, we amended all commit messages accordingly.

File: nestkernel/connection_manager.cpp (hunk 1 above)
Comment (jasperalbers) [reply nestkernel/connection_manager.cpp]: Thanks, this seems to have solved the issue!

**Review by stinebuu — APPROVED** [review]
:+1:

**Review by hakonsbm — APPROVED** [review]
LGTM!

**Pull request merged**